app/models/submission.rb

```diff
@@ -32,5 +34,14 @@ def reject
   def average_rate
     rates.count == 0 ? 0 : (rates.sum(:value).to_f / rates.count)
   end
-end
 
+  def generate_confirmation_token!
+    self.confirmation_token = Devise.friendly_token
+    self.confirmation_token_created_at = Time.now
+    save
+  end
+
+  def has_expired?
+    awaiting? && confirmation_token_created_at < 1.week.ago
+  end
```

Comment:
@marekciupak @ArturT @czorek 
past_confirmation_due_date?
past_confirmation_period?

The 'confirmation' term makes some trouble with wording, maybe it would be better to call the status "invitation_status" instead, although I wouldn't make such a change until we consider @marekciupak's suggestion about dealing with this without additional status whatsoever.

to_be_marked_as_expired? - this method is used for more than that